azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -0,0 +1,342 @@
+import com.microsoft.azure.management.Azure.Authenticated;
+import com.microsoft.azure.management.microservices4spring.v2019_05_01_preview.implementation.AppClusterResourceInner;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.maven.spring.configuration.Deployment;
+import com.microsoft.azure.maven.spring.exception.NoResourcesAvailableException;
+import com.microsoft.azure.maven.spring.prompt.IPrompter;
+import com.microsoft.azure.maven.spring.spring.SpringServiceUtils;
+import com.microsoft.azure.maven.spring.utils.MavenUtils;
+import com.microsoft.azure.maven.spring.utils.Utils;
+import com.microsoft.azure.maven.spring.utils.XmlUtils;
+import com.microsoft.azure.maven.utils.TextUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.model.Build;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.io.jdom.MavenJDOMWriter;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.IOUtil;
+import org.codehaus.plexus.util.WriterFactory;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.jdom2.Document;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.Format;
+import org.jdom2.output.Format.TextMode;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Mojo(name = "config", requiresProject = true, requiresDirectInvocation = true, aggregator = true)
+public class ConfigMojo extends AbstractSpringMojo {
+    private static final String DEPLOYMENT_TAG = "deployment";
+
+    @Parameter(property = "full")
+    private boolean full;
+
+    private IPrompter prompt = null; // new DefaultPrompter();
+
+    private List<MavenProject> targetProjects = new ArrayList<>();
+
+    private List<MavenProject> publicProjects;
+
+    private Authenticated azure;
+
+    @Override
+    protected void doExecute() throws MojoExecutionException, MojoFailureException {
+        if (!settings.isInteractiveMode()) {
+            throw new MojoFailureException("The goal 'config' must be run at interactive mode.");
+        }
+
+        if (!Utils.isPomPackagingProject(this.project) && !Utils.isJarPackagingProject(this.project)) {
+            throw new MojoFailureException(String.format("The project (%s) with packaging %s is not supported for azure spring cloud service.",
+                    this.project.getName(), this.project.getPackaging()));
+        }
+        if (isProjectConfigured(this.project)) {
+            throw new MojoFailureException(String.format("Project (%s) is already configured.", this.project.getName()));
```

Comment:
Should we throw an Exception here? Or just warning/info and return?

Reply:
In the spec, it is considered as error, because we did nothing, a failure will enforce user to check the error details.